src/main/controllers/features/sole-trader/typeOfBusinessController.ts

```diff
@@ -67,30 +76,46 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
             if (selectedOption !== "OTHER") {
                 const acspData : ACSPData = {
                     id: email,
-                    typeofBusiness: selectedOption
+                    typeOfBusiness: selectedOption
                 };
-                saveDataInSession(req, USER_DATA, acspData);
+                saveDataInSession(req, USER_DATA, acspData); // should be removed when refactoring
                 const answersArray: Answers = {
-                    typeofBusiness: TypeOfBusiness[selectedOption as keyof typeof TypeOfBusiness]
+                    typeOfBusiness: TypeOfBusiness[selectedOption as keyof typeof TypeOfBusiness]
                 };
                 saveDataInSession(req, ANSWER_DATA, answersArray);
             }
-
-            switch (selectedOption) {
-            case "LIMITED_COMPANY":
-            case "LIMITED_PARTNERSHIP":
-            case "LIMITED_LIABILITY_PARTNERSHIP":
-                res.redirect(addLangToUrl(BASE_URL + LIMITED_WHAT_IS_THE_COMPANY_NUMBER, lang));
-                break;
-            case "PARTNERSHIP":
-                res.redirect(addLangToUrl(BASE_URL + UNINCORPORATED_NAME_REGISTERED_WITH_AML, lang));
-                break;
-            case "SOLE_TRADER":
-                res.redirect(addLangToUrl(BASE_URL + SOLE_TRADER_WHAT_IS_YOUR_ROLE, lang));
-                break;
-            case "OTHER":
-                res.redirect(addLangToUrl(BASE_URL + OTHER_TYPE_OF_BUSINESS, lang));
-                break;
+            const acspData: ACSPData = session.getExtraData(USER_DATA)!;
+            const acsp: Acsp = {
```

Comment:
Does this need to be ACSP type? or can it be ACSPData type? and then we can use the acspData directly

Reply:
Yes, As per current implementation of API client it needs to be ACSP type. We can change client implementation in api-sdk-node and acsp-api if we want otherwise.